scio-core/src/main/scala/com/spotify/scio/coders/instances/AvroCoders.scala

```diff
@@ -164,7 +165,7 @@ trait AvroCoders {
       override def apply(writer: Schema): DatumWriter[T] = {
         // create the datum writer using the schema api
         // class API might be unsafe. See schemaForClass
-        val datumWriter = new ReflectDatumWriter[T](schemaForClass(clazz).get)
+        val datumWriter = new ReflectDatumWriter[T](schemaForClass(clazz).get, new ReflectData())
```

Comment:
```suggestion
        val datumWriter = new ReflectDatumWriter[T](writer, new ReflectData())
```
and remove `setSchema`